src/Compilers/Core/Portable/SourceGeneration/SourceGeneratorContext.cs

```diff
@@ -39,11 +39,27 @@ internal SourceGeneratorContext(Compilation compilation, AnalyzerOptions options
         public void ReportDiagnostic(Diagnostic diagnostic) { throw new NotImplementedException(); }
     }
 
+    [System.Diagnostics.CodeAnalysis.SuppressMessage("ApiDesign", "RS0016:Add public types and members to the declared API", Justification = "In Progress")]
+    public struct InitializationContext
+    {
+        internal InitializationContext(CancellationToken cancellationToken = default)
+        {
+            CancellationToken = cancellationToken;
+            InfoBuilder = new GeneratorInfo.Builder();
+        }
+
+        public CancellationToken CancellationToken { get; }
+
+        internal GeneratorInfo.Builder InfoBuilder { get; }
```

Comment:
>GeneratorInfo.Builder InfoBuilder [](start = 17, length = 33)

Will this field grow to include more items? If not, consider replacing with single `EditCallback` property.

Reply:
Yes I'm expecting it to add a bunch more as we go forward.

---
In reply to: [392538403](https://github.com/dotnet/roslyn/pull/42359#discussion_r392538403) [](ancestors = 392538403)